pbsmetrics/go_metrics.go

```diff
@@ -313,6 +348,22 @@ func (me *Metrics) RecordAdapterBidsReceived(labels AdapterLabels, bids int64) {
 	aam.BidsReceivedMeter.Mark(bids)
 }
 
+// RecordAdapterBidAdm implements a part of the MetricsEngine interface.
+// This tracks how many bids from each Bidder use `adm` vs. `nurl.
+func (me *Metrics) RecordAdapterBidAdm(labels AdapterLabels, bidType openrtb_ext.BidType, hasAdm bool) {
+	am, ok := me.AdapterMetrics[labels.Adapter]
+	if !ok {
+		glog.Errorf("Trying to run adapter bid metrics on %s: adapter metrics not found", string(labels.Adapter))
+		return
+	}
+	if hasAdm {
+		am.MarkupMetrics[bidType].AdmMeter.Mark(1)
```

Comment:
We should do an ok type check on `am.MarkupMetrics[bidType]`. Today there may only be the 4 options, but if someone adds a fifth type of bid into `openrtb_ext.BidType`, it will not be obvious that this piece of code will also need to be touched. Will probably take some time to track down the cause of infrequent crashes, as a new type will rarely be used initially.

Reply:
I'm a little torn on this... it's something which _could_ be checked before runtime, but it's painful to do so (parse the source code for BidTypes? Add a `BidTypes() []BidType` function to `openrtb_ext` which we use in the test which returns all of them?)

This works as well as anything else, though.